Update installer script and italian language

bovirus opened the pull request:
@az0 

Add variable for copyright name
Add BrandingText setting (use copyright name instead "NSIS v. 3.03")
Add text string for DetailPrint (string1 and 2)

- Improvements for next version
Localization of section name and section description

- Issue to fix (present in original script)
In Windows program list the program name is showed as program name + version.
Should be only program name.
I didn't know where is set.

Please check and merge.

bovirus commented:
@az0 

Please merge this. Thanks.
Then in next step I will test the translation for section and section description.

bovirus commented:
@az0 

Pending PR request.

az0 commented:
@bovirus  

It is difficult to review because it looks like 100% of the lines were changed, even the lines not related to the changes you described. Are you able to adjust this please?

bovirus commented:
@az0 

Please take care that is not 100% changed but lot of less.

In main installer script I didn't change anything about code but I made some changes only in two specific areas.

In [Setup] section I grouped the settings to be more clear and self documneted.
And added some varibales that was not present.

In other part of main script I didn't touch the code but I replaced hard coded text  message (in english) with variables (ex. detailPrint, sectiond escription) that point to a string that can be translated.
For me the script as much as possible shouldn't contain hard coded english text string.

I didn't delete anything.
I just add something because it was not exist before.

Witrh my proposal I added som imporvements (nothing less before).

bovirus commented:
** Change1

  !define PROG_AUTHOR "Andrew Ziem"
  !define PROG_COPYRIGHT "Andrew Ziem"
  BrandingText "${PROG_COPYRIGHT}"
  Name "${prodname} ${VERSION}"

  ; ----------------------------- disable warning overwrite default language strings (wrn 6030)
  ;!pragma warning disable 6030
  ; ----------------------------- disable warning missing strings in some not english section (wrn 6040)
  !pragma warning disable 6040

- Add variable for copyright author, program copyright
- Change branding etxt from NSISI fro the program author
- Change top left messagge in main window from prog name to progname + version (now you can see version that you are  installing)

** Change2

!ifdef NoTranslations
  VIAddVersionKey /LANG=1033 "ProductName"     "${prodname}"
  VIAddVersionKey /LANG=1033 "ProductVersion"  "${File_VERSION}"
  VIAddVersionKey /LANG=1033 "Comments"        ""
  VIAddVersionKey /LANG=1033 "CompanyName"     "BleachBit.org"
  VIAddVersionKey /LANG=1033 "LegalTrademarks" "${PROG_AUTHOR}"
  VIAddVersionKey /LANG=1033 "LegalCopyright"  "${PROG_COPYRIGHT}"
  VIAddVersionKey /LANG=1033 "FileVersion"     "${File_VERSION}"
  VIAddVersionKey /LANG=1033 "FileDescription" "${prodname} Setup"
!endif

!ifndef NoTranslations
  VIAddVersionKey /LANG=0    "ProductName"     "${prodname}"
  VIAddVersionKey /LANG=0    "ProductVersion"  "${File_VERSION}"
  VIAddVersionKey /LANG=0    "Comments"        ""
  VIAddVersionKey /LANG=0    "CompanyName"     "BleachBit.org"
  VIAddVersionKey /LANG=0    "LegalTrademarks" "${PROG_AUTHOR}"
  VIAddVersionKey /LANG=0    "LegalCopyright"  "${PROG_COPYRIGHT}"
  VIAddVersionKey /LANG=0    "FileVersion"     "${File_VERSION}"
  VIAddVersionKey /LANG=0    "FileDescription" "${prodname} Setup"
!endif

; VIAddVersionKey "InternalName"       ""
; VIAddVersionKey "OriginalFilename"   ""

- Add legalTrademake and "comments fiule properties settings
- Grouping file properteis values for tipology
- Add comneted variable about internal name/original name

** Change 3

;!define MUI_FINISHPAGE_LINK "Visit the ${prodname} web site."
!define MUI_FINISHPAGE_LINK "$(BLEACHBIT_MUI_FINISHPAGE_LINK)"
!define MUI_FINISHPAGE_LINK_LOCATION "https://www.bleachbit.org"

DetailPrint "$(MULTIPRINT1)"
DetailPrint "$(MULTIPRINT2)"

- replace hard coded text with a text variable

Change 4 (section area)

SectionGroup /e "$(SECTION_SHORTCUTS)" SECTION_SHORTCUTS
    Section "$(SECTION_START_MENU)" SECTION_START_MENU

and the same for other sction
- I changed fixed section name in english with a variable that enable section name tranmslation.
 
I hope that it explain enough.

az0 commented:
On the [files tab](https://github.com/bleachbit/bleachbit/pull/1461/files) of this PR, every line of code is modified: 2623 lines added and 2594 lines removed, so perhaps there is a difference of whitespace or line endings (like CR/LF vs LF).

sanjaykaria commented:
> On the [files tab](https://github.com/bleachbit/bleachbit/pull/1461/files) of this PR, every line of code is modified: 2623 lines added and 2594 lines removed, so perhaps there is a difference of whitespace or line endings (like CR/LF vs LF).

The setting highlighted in attached screenshot might help reduce the differences.
![image](https://user-images.githubusercontent.com/3830310/216835808-1346e42a-190a-4fad-8d55-87de88d00610.png)

bovirus commented:
@az0 

Sorry for the problem.
Was an error in my files that has CR-LF termination instead LF (as original).
I uplaoded new file with correct only LF termination.
Please check.

bovirus commented:
@az0 

The ending termination text problem was fixed.

Could you check please?

sanjaykaria commented on windows/NsisInclude/NsisMultiUserLang.nsh:
```diff
@@ -1,2101 +1,2115 @@
+	LangString UNINSTALL_TEXT ${LANG_ENGLISH} "BleachBit will be uninstalled from the following folder.  Click Uninstall to start the uninstallation.  WARNING: The uninstaller completely removes the installation directory including any files (such as custom cleaners) that you may have added or changed."
+	LangString ALREADY_INSTALLED ${LANG_ENGLISH} "${prodname} is already installed.  Click 'OK' to uninstall the old version before upgrading, or click 'Cancel' to abort the upgrade."
+	; shred shell integration menu name
+	LangString SHRED_SHELL_MENU ${LANG_ENGLISH} "Shred with Beachblit"
```
Should be "Shred with BleachBit"

bovirus replied:
You are right. Sorry for mismatch.

sanjaykaria commented on windows/bleachbit.nsi:
```diff
@@ -1,493 +1,508 @@
+  Name "${prodname} ${VERSION}"
+
+  ; ----------------------------- disable warning overwrite default language strings (wrn 6030)
+  ;!pragma warning disable 6030
```
It might be a good idea to push the warning and then after disabling the needed warning, pop back warning.

Reference: https://stackoverflow.com/questions/4193476/is-using-pragma-warning-push-pop-the-right-way-to-temporarily-alter-warning-lev

(Although the above is for C++, I guess the same would apply for NSIS, too)

bovirus replied:
@sanjaykaria

The disabled warning  are only about missing strings in multiusernsh  file could be disabled AFTER will aligned all languages with all strings.
Otherwise you will have a lot of warnings that can create misunderstanding.

sanjaykaria replied:
@bovirus 
I understand the need of disabling the warning, but what am referring to is just saving the state of warning (in case it applies in NSIS too, as it does in C++) and then restoring the state back. 
I am not an expert in NSIS, but just a thought.

bovirus replied:
@ sanjaykaria
For this moment to go ahead I decide to leave it as in original without warning but with comm,ent it hen if you need you can enable and remind the use.

sanjaykaria commented on windows/NsisInclude/NsisMultiUserLang.nsh:
```diff
@@ -1,2101 +1,2115 @@
+	LangString MULTIUSER_ADMIN_CREDENTIALS_REQUIRED ${LANG_CROATIAN} "Administrator credentials required."
+	LangString MULTIUSER_ADMIN_UNINSTALL_CREDENTIALS_REQUIRED ${LANG_CROATIAN} "Administrator credentials required for uninstall."
+	; error messages - not so important
+	LangString MULTIUSER_I
```
Should be 
.... "LangString MULTIPRINT2 ${**LANG**_ITALIAN}" ....

bovirus replied:
Thanks. Fixed.

sanjaykaria commented:
> ******************** Change 4 (section area)
> 
> SectionGroup /e "$(SECTION_SHORTCUTS)" SECTION_SHORTCUTS Section "$(SECTION_START_MENU)" SECTION_START_MENU
> 
> and the same for other sctions
> 
> * I changed fixed section name in english with a variable that enable section name tranmslation.

@bovirus 
As far as I understand from the code changes, it seems that Section names which seems to have been translated before this changes have been hard-coded to English. 
![image](https://user-images.githubusercontent.com/3830310/219351329-f0cc186a-c367-45f6-8c0f-7a9e1e99c447.png)


Not sure if am missing something here, but this change will show the component name in in English instead of Italian, as below, .
![image](https://user-images.githubusercontent.com/3830310/219351804-73afe9ff-ee76-4947-ab4f-c9888ed4b653.png)

az0 commented:
@bovirus  @sanjaykaria  Thank you both for working on this!

bovirus commented:
@sanjaykaria

How did you build the installer?

sanjaykaria commented:
> @sanjaykaria
> 
> How did you build the installer?

@bovirus 

When a pull request is created ( or modified ) the Appveyor build system fires a build (on remote VM) with the latest code change of the pull request ( as an automated pre-check ). The build artifacts are uploaded after a successful build. I used the installer from this build artifacts.

The builds from commits are available at 
https://ci.appveyor.com/project/az0/bleachbit/history
